source/matplot/core/axis_type.cpp

```diff
@@ -257,7 +257,11 @@ namespace matplot {
                     char buff[20] = {0};
                     time_t now = (time_t)tick_values_[i];
                     struct tm buf;
+#if defined(__unix__)
+                    localtime_r(&now, &buf);
+#elif defined(_WIN32)
                     localtime_s(&buf, &now);
+#endif
```

Comment:
Nice. We need a third alternative when both conditions are false.

```cpp
#if defined(__unix__)
                    localtime_r(&now, &buf);
#elif defined(_WIN32)
                    localtime_s(&buf, &now);
#else
                    // use std::localtime from https://en.cppreference.com/w/cpp/chrono/c/localtime
#endif
```

We also need an equivalent pattern to include the proper headers:

```cpp
#if defined(__unix__)
#include <header with localtime_r>
#elif defined(_WIN32)
#include <header with localtime_s>
#else
#include <ctime>
#endif
```